privacyidea/lib/importotp.py

```diff
@@ -529,10 +529,15 @@ def parsePSKCdata(xml_data,
 
         parameters = key.algorithmparameters
         token["otplen"] = parameters.responseformat["length"] or 6
-        try:
-            token["hashlib"] = parameters.suite["hashalgo"] or "sha1"
-        except Exception as exx:
-            log.warning("No compatible suite contained.")
+        # token["hashlib"] = parameters.suite or "sha1"
+
+        hash_lib = "sha1"
+
+        # Check if hashlib is explicitly set in file
+        if parameters.suite and parameters.suite.string:
+            hash_lib = parameters.suite.string.lower()
```

Comment:
Is the name of the hash algorithm contained in ``parameters.suite["hashalgo"]`` or in ``paramters.suite.string``?
This looks like a change here. I was looking for quite some time myself, but did not find it.
Also, in regards to the problem, that other vendors are not quite precise in regards to pskc, I would wrap line 537 and 538 still in an try-except block.

Reply:
I have described this change / why I changed this [here](https://github.com/privacyidea/privacyidea/issues/2257#issuecomment-777491824) in more detail. I.e., the only reference I found says that it is contained in the `string` property.

But I will wrap it just to be sure.